Userland/Libraries/LibCore/Stream.cpp

```diff
@@ -170,6 +170,29 @@ ErrorOr<Bytes> File::read(Bytes buffer)
     return buffer.trim(nread);
 }
 
+ErrorOr<ByteBuffer> File::read_all(size_t block_size)
+{
+    // Note: This is used as a heuristic, it's not valid for devices or /proc files.
+    auto potential_file_size = TRY(System::fstat(m_fd)).st_size;
+
+    ByteBuffer data;
+    data.ensure_capacity(potential_file_size);
+
+    size_t total_read {};
+    while (true) {
+        auto chunk = TRY(data.get_bytes_for_writing(block_size));
+        auto const nread = TRY(read(chunk)).size();
+
+        total_read += nread;
+        if (nread < block_size)
```

Comment:
nit: this does not necessarily indicate that the file is at EOF. Only read() == 0 does (or, better yet File::is_eof())